src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -381,7 +383,12 @@ private Object readResolve() {
         public @CheckForNull Run<?,?> runForDisplay() {
             Run<?,?> r = run();
             if (r == null && /* not stored prior to 1.13 */runId != null) {
-                return Run.fromExternalizableId(runId);
+                SecurityContext orig = ACL.impersonate(ACL.SYSTEM);
+                try {
+                    return Run.fromExternalizableId(runId);
```

Comment:
Not to worry about showing the task to other users in the executor widget; [this display code](https://github.com/jenkinsci/workflow-durable-task-step-plugin/blob/1396e6fb8fbbd1ac97a68dfc9b884706d4fc5b59/src/main/resources/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution/PlaceholderTask/PlaceholderExecutable/executorCell.jelly#L31-L38) already handled that, since `run()` will _normally_ return something without an access check (while the `node` block is up and running)—this code path is only used during `ExecutorPickle` rehydration when resuming a build.